indico/modules/users/export_schemas.py

```diff
@@ -0,0 +1,398 @@
+                  'submission_comment', 'submitted_dt', 'state', 'url', 'event_url')
+
+    description = String(attribute='_description')
+    state = Enum(AbstractState)
+    url = Function(lambda abstract: url_for('abstracts.display_abstract', abstract, management=False, _external=True))
+    event_url = Function(lambda abstract: url_for('events.display', abstract.event, _external=True))
+    files = List(Nested(AbstractFileExportSchema))
+
+
+class PaperFileExportSchema(mm.SQLAlchemyAutoSchema):
+    class Meta:
+        model = PaperFile
+        fields = ('id', 'filename', 'md5', 'url', 'path')
+
+    url = Function(lambda f: url_for('papers.download_file', f.paper.contribution, f, _external=True))
+    path = Function(lambda file: build_storage_path(file))
+
+
+class PaperExportSchema(mm.SQLAlchemyAutoSchema):
+    class Meta:
+        model = Contribution
+        fields = ('id', 'title', 'description', 'submitted_dt', 'url', 'event_url', 'files')
+
+    submitted_dt = Method('_get_submitted_dt')
+    state = Enum(PaperRevisionState)
+    url = Function(lambda paper: url_for('papers.paper_timeline', paper, _external=True))
+    event_url = Function(lambda paper: url_for('events.display', paper.event, _external=True))
+    files = List(Nested(PaperFileExportSchema))
+
+    def _get_submitted_dt(self, paper):
+        return DateTime().serialize('submitted_dt', paper.revisions[0])
+
+
+class PersonalDataExportSchema(mm.SQLAlchemyAutoSchema):
+    class Meta:
+        model = User
+        fields = ('first_name', 'last_name', 'full_name', 'title', 'email', 'secondary_emails',
+                  'affiliation', 'phone', 'avatar_url', 'picture_metadata', 'picture_source',
+                  'favorite_events', 'favorite_categories')
+
+    title = NoneValueEnumField(UserTitle, none_value=UserTitle.none, attribute='_title')
+    secondary_emails = Function(lambda user: list(user.secondary_emails))
+    affiliation = Nested(AffiliationSchema, attribute='affiliation_link')
+    favorite_events = List(Nested(EventExportSchema))
+    favorite_categories = List(Nested(CategoryExportSchema))
+
+
+class RoomBookingExportSchema(Schema):
+    from indico.modules.rb.schemas import ReservationSchema
+
+    favorite_rooms = List(Nested(RoomExportSchema))
+    owned_rooms = List(Nested(RoomExportSchema))
+    reservations = List(Nested(ReservationSchema))
+
+
+class MiscDataExportSchema(Schema):
+    created_events = List(Nested(EventExportSchema))
+    static_sites = List(Nested(StaticSiteExportSchema))
+    identities = List(Nested(IdentityExportSchema))
+    personal_tokens = List(Nested(PersonalTokenExportSchema))
+    api_key = Nested(APIKeyExportSchema)
+    old_api_keys = List(Nested(APIKeyExportSchema))
+    oauth_applications = Method('_serialize_oauth_apps')
+
+    def _serialize_oauth_apps(self, user):
+        oauth_apps = [link.application for link in user.oauth_app_links]
+        return OAuthApplicationExportSchema(many=True).dump(oauth_apps)
```

Comment:
I think exporting the scopes on the link itself would be good, since that's the scope the application currently has for this particular user.

Something like this (ie using a `List(Nested(OAuthApplicationUserLinkExportSchema))`

```yaml
oauth_applications:
- scopes: ...
  application:
    name: ...
    description: ...
```